### ISSUE TITLE
Extension name with more than 20 characters throws error

### ISSUE THREAD

**Issue by deissnerk** [cloudevents/sdk-go#670]
[`SetExtension()`](https://github.com/cloudevents/sdk-go/blob/f322543770b0da85f3e001dbda6f9e686cbe745f/v2/event/eventcontext_v1.go#L75) of the v1.0 event context throws an error, if the name of the extension is longer than 20 characters. In the spec this length restriction is a [SHOULD NOT](https://github.com/cloudevents/spec/blob/v1.0.1/spec.md#attribute-naming-convention) restriction. So failing with an error seems to be too strong. Shouldn't it be possible to violate SHOULD requirements? 😉

**Comment by n3wscott** [comment]
It should be possible. Good point. What version of the sdk are you seeing this in? it is in the latest 2.4.0?

**Comment by deissnerk** [comment]
We experienced it first with the 2.4.0 release. I think it might have sneaked in with this [commit](https://github.com/cloudevents/sdk-go/commit/2c1245bfced37e33cc33c00d1abadb8d11f1a63f).

**Comment by n3wscott** [comment]
Yep. ok thanks! I will get a point fix release out in the next couple days.

**Comment by deissnerk** [comment]
Thank you. If it is only about removing the length check, I can come up with a little PR, too.

**Comment by n3wscott** [comment]
@deissnerk If you want to write the PR, that would be great, I was thinking of having a global setting that an importer can set the max length to something they wish, but have the default be unlimited (0).
